pyiron_base/generic/inputlist.py

```diff
@@ -1,656 +1,5 @@
-            group_name (str, optional): HDF5 subgroup name, overrides
-                self.table_name
-        """
-
-        group_name = group_name or self.table_name
-        if group_name:
-            hdf = hdf.create_group(group_name)
-
-        self._type_to_hdf(hdf)
-        hdf["data"] = self.to_builtin()
-        hdf["read_only"] = self.read_only
-
-    def _type_to_hdf(self, hdf):
-        """
-        Internal helper function to save type and version in hdf root
-
-        Args:
-            hdf (ProjectHDFio): HDF5 group object
-        """
-        hdf["NAME"] = self.__class__.__name__
-        hdf["TYPE"] = str(type(self))
-        hdf["VERSION"] = self.__version__
-        hdf["OBJECT"] = "InputList"
-
-    def from_hdf(self, hdf, group_name=None):
-        """
-        Restore the InputList from an HDF5 file.  If group_name or
-        self.table_name are not None, open a sub group in hdf prior to reading
-        if not read directly from hdf.  group_name overrides self.table_name if
-        both are not None.
-
-        Args:
-            hdf (ProjectHDFio): HDF5 group object
-            group_name (str, optional): HDF5 subgroup name, overrides
-                self.table_name
-        """
-
-        group_name = group_name or self.table_name
-        if group_name:
-            with hdf.open(group_name) as hdf_group:
-                data = hdf_group["data"]
-                if "read_only" in hdf_group.list_nodes():
-                    read_only = hdf_group["read_only"]
-                else:
-                    read_only = False
-        else:
-            data = hdf["data"]
-            if "read_only" in hdf.list_nodes():
-                read_only = hdf["read_only"]
-            else:
-                read_only = False
-
-        self.clear()
-
-        self.update(data, wrap=True)
-        self.read_only = bool(read_only)
-
-    def nodes(self):
-        """
-        Iterator over keys to terminal nodes.
-
-        Returns:
-            :class:`list`: list of keys to normal values.
-        """
-        for k, v in self.items():
-            if not isinstance(v, InputList):
-                yield k
-
-    def list_nodes(self):
-        """
-        Return a list of keys to terminal nodes.
-
-        Returns:
-            :class:`list`: list of keys to normal values.
-        """
-        return list(self.nodes())
-
-    def groups(self):
-        """
-        Return a list of keys to nested lists.
-
-        Returns:
-            :class:`list`: list of all keys to elements of :class:`InputList`.
-        """
-        for k, v in self.items():
-            if isinstance(v, InputList):
-                yield k
-
-    def list_groups(self):
-        """
-        Return a list of keys to nested lists.
-
-        Returns:
-            :class:`list`: list of all keys to elements of :class:`InputList`.
-        """
-        return list(self.groups())
+from .datacontainer import DataContainer as InputList
```

Comment:
I would derive a class named `InputList` from the `DataContainer`, then once this class is used we can print a warning.